src/queue.cpp

```diff
@@ -399,6 +399,20 @@ cl_int cvk_command_kernel::build() {
                            &m_global_offsets);
     }
 
+    if (m_kernel->has_pod_arguments() &&
+        !m_kernel->has_pod_buffer_arguments()) {
+        for (auto& arg : m_kernel->arguments()) {
+            if (arg.kind == kernel_argument_kind::pod_pushconstant) {
+                CVK_ASSERT(arg.offset + arg.size <=
+                           m_argument_values->pod_pushconstant_buffer().size());
+                vkCmdPushConstants(
```

Comment:
The corresponding ranges have not be declared at pipeline layout creation time (see `m_push_constant_ranges`). This probably won't pass validation.

Reply:
Thanks, I've added this in the latest commit.